pkg/cri/resource-manager/cache/pod.go

```diff
@@ -109,6 +111,32 @@ func (p *pod) GetContainers() []Container {
 	return containers
 }
 
+// Get container pointer by its name.
+func (p *pod) getContainer(name string) *container {
+	var found *container
+
+	if p.containers == nil {
+		p.containers = make(map[string]string)
+	}
+
+	for _, c := range p.GetContainers() {
+		cptr := c.(*container)
+		p.containers[cptr.Name] = cptr.ID
+		if cptr.Name == name {
+			found = cptr
+		}
+	}
+
+	return found
+}
+
+// Get container by its name.
+func (p *pod) GetContainer(name string) (Container, bool) {
+	c := p.getContainer(name)
+
+	return c, c != nil
```

Comment:
Again, was there a reason that we want to return two values?

Reply:
We want to return one interface value that we can tell apart from a nil Container without using reflect... ;-)